### PR TITLE
Handle GET /api/v1/flatsUsers

### PR DESCRIPTION
**Pull request by bayan2002**

### CODE REVIEW

File: server/controllers/flatsUsers/getFlatsUsers.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  try {
+    const { order } = req.query;
+    const sortingOrder = (order === 'descend') ? 'DESC' : 'ASC';
+
+    const limitNum = 2;
+    const { page } = req.query;
+    const total = await FlatModel.count({ include: [{ model: UserModel }] });
+    if ((+page) <= 0) {
+      throw new CustomError(400, 'bad request');
+    }
```
Comment (Mu7ammadAbed) [inline server/controllers/flatsUsers/getFlatsUsers.ts:17]:
```suggestion
    if (!(Number(page) > 0)) {
      throw new CustomError(400, 'Query Parameter Page is required and must be a number greater than 0');
    }
```

File: server/controllers/flatsUsers/getFlatsUsers.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  phone_number: string;
+}
+interface IQueryResult {
+  id: number;
+  flat_number: number;
+  // 'User.first_name': string;
+  // 'User.last_name': string;
+  // 'User.phone_number': string;
+  // 'User.id': number;
+  // 'User.email': string;
```
Comment (Mu7ammadAbed) [inline server/controllers/flatsUsers/getFlatsUsers.ts:60]:
```suggestion
```

File: server/routes/flatsUsersRouter.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { Router } from 'express';
+import { getFlatsUsers } from '../controllers/index';
+
+const router = Router();
+
+router.get('/', getFlatsUsers);
```
Comment (Mu7ammadAbed) [inline server/routes/flatsUsersRouter.ts:6]: move this route to the flats route file

File: server/controllers/flatsUsers/getFlatsUsers.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import { CustomError } from '../../helpers';
+import { FlatModel, UserModel } from '../../models';
+
+export default async (req:Request, res:Response, next:NextFunction) => {
+  try {
+    const { order } = req.query;
+    const sortingOrder = (order === 'descend') ? 'DESC' : 'ASC';
+
+    const limitNum = 2;
+    const { page } = req.query;
```
Comment (Mu7ammadAbed) [inline server/controllers/flatsUsers/getFlatsUsers.ts:13]:
```suggestion
    const { order, page } = req.query;
    const sortingOrder = (order === 'descend') ? 'DESC' : 'ASC';

    const limitNum = 2;
```

**Review by mohjaps — APPROVED** [review]

**Review by Mai-Elhajeen — APPROVED** [review]

**Review by tareq-abuaqlain — APPROVED** [review]

**Pull request merged**